code/scripting/api/objs/ship.cpp

```diff
@@ -1853,6 +1853,31 @@ ADE_FUNC(getDisplayString, l_Ship, nullptr, "Returns the string which should be
 	return ade_set_args(L, "s", shipp->get_display_name());
 }
 
+ADE_FUNC(vanish, l_Ship, nullptr, "Vanishes this ship from the mission. Works in Singleplayer only and will cause the ship exit to not be logged.", nullptr, "Returns nothing.")
+{
+
+	object_h* objh = nullptr;
+	ship* shipp = nullptr;
+
+	if (!ade_get_args(L, "o", l_Ship.GetPtr(&objh)))
+		return ADE_RETURN_NIL;
+
+	if (!objh->IsValid())
+		return ADE_RETURN_NIL;
+
+	shipp = &Ships[objh->objp->instance];
+
+	// if MULTIPLAYER bail
+	if (Game_mode & GM_MULTIPLAYER)
+		return ADE_RETURN_NIL;
+
+	auto ship_entry = &Ship_registry[Ship_registry_map[shipp->ship_name]];
```

Comment:
`objh->objp->instance` will also give you the ship number so you can avoid going through the registry map to get that.

Reply:
Is the Ship_registry index guaranteed to be equal to the Ship's instance number in the Ships array?
Was not sure of this since other ship functions to this as well as opposed to using the the instance number when accessing ShipStatus.

Reply:
Yes, I was mistaken here.

However, you do not need to go through the registry at all here. If you have a valid object handle (which is the case at this point) then you know that the ship is present and you can call `ship_actually_depart` unconditionally.

Reply:
Ah okay. That explains why the sexp needs to do that check but the API does not.
Thanks for clarifying